packages/project/src/db/props.ts

```diff
@@ -1,22 +1,75 @@
 import {
   type Tree,
   type AllUserProps,
-  UserProps,
+  type UserProp,
+  // TODO: move zod schema into project package
+  zUserUserProps,
+  zUserDbProps,
 } from "@webstudio-is/react-sdk";
 import { applyPatches, type Patch } from "immer";
 import { prisma } from "@webstudio-is/prisma-client";
+import { formatAsset } from "@webstudio-is/asset-uploader/server";
 
 export const loadByTreeId = async (treeId: Tree["id"]) => {
   const instancePropsEntries = await prisma.instanceProps.findMany({
     where: { treeId },
   });
 
+  // Fin all asset_id in all props
+  const assets_id: string[] = [];
+
+  for (const instanceProps of instancePropsEntries) {
+    const props = JSON.parse(instanceProps.props);
+    const dpProps = zUserDbProps.parse(props);
+
+    for (const dbProp of dpProps) {
+      if ("asset_id" in dbProp) {
+        assets_id.push(dbProp.asset_id);
+      }
+    }
+  }
+  // Load all assets
+  const assets = await prisma.asset.findMany({
+    where: {
+      id: {
+        in: assets_id,
+      },
+    },
+  });
+
+  const assetsMap = new Map(
+    assets.map((asset) => [asset.id, formatAsset(asset)])
+  );
+
   return instancePropsEntries.map((instanceProps) => {
     const props = JSON.parse(instanceProps.props);
-    UserProps.parse(props);
+    const resProps: UserProp[] = [];
+    // Can be changed onto `props as ` having that above is already checked, no optmizations now
+    const dpProps = zUserDbProps.parse(props);
+
+    for (const dbProp of dpProps) {
+      if ("asset_id" in dbProp) {
+        const asset = assetsMap.get(dbProp.asset_id);
+        if (asset) {
+          resProps.push({
+            id: dbProp.id,
+            prop: dbProp.prop,
+            asset,
+            required: dbProp.required,
+          });
+        } else {
+          // Can be ok if asset was deleted having that no db constraint checks exists
+          // TODO: remove throw on something else
+          throw new Error(`asset with ${dbProp.asset_id} not found`);
```

Comment:
just fui for consistent error style
```suggestion
          throw new Error(`Asset with assetId "${dbProp.asset_id}" not found`);
```